agent/js/package.json

```
@@ -1,24 +1,45 @@
 {
-  "name": "agent-sdk",
+  "name": "agent-protocol",
   "version": "0.1.0",
   "description": "API for interacting with Agent",
+  "main": "./dist/index.js",
+  "module": "./dist/index.m.js",
+  "unpkg": "dist/index.umd.js",
+  "types": "dist/index.d.ts",
   "scripts": {
     "prepublishOnly": "pnpm build",
-    "build": "rm -rf ./dist && tsup --minify",
-    "dev": "tsup --watch",
-    "fix": "npx eslint **/src/**/*.{ts,js} --fix",
+    "clean": "rm -rf ./dist",
+    "build": "rm -rf ./dist && microbundle --tsconfig tsconfig.json --no-sourcemap -f cjs,umd src/index.cjs.ts && microbundle --tsconfig tsconfig.json --no-sourcemap -f modern,esm src/index.ts && mv dist/src/* dist/ && rm -rf dist/{examples,src}",
```

```suggestion
    "build": "rm -rf ./dist && tsup",
```

with this config file works well for me 
```js
import { defineConfig } from 'tsup'

import * as packageJSON from './package.json'

const excludedPackages = []

export default defineConfig({
  entry: ['src/index.ts'],
  target: 'node16',
  platform: 'node',
  format: 'cjs'
})
```